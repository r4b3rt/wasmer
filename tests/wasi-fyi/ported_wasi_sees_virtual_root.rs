// WASI:
// mapdir: act1:test_fs/hamlet/act1
// mapdir: act2:test_fs/hamlet/act2
// mapdir: act1-again:test_fs/hamlet/act1

use std::fs;

fn main() {
    let results = {
        let mut out = vec![];

        let read_dir = fs::read_dir("/").unwrap();
        for entry in read_dir {
            out.push(format!("{:?}", entry.unwrap().path()))
        }
        let read_dir = fs::read_dir("/hamlet/act1/..").unwrap();
        for entry in read_dir {
            out.push(format!("{:?}", entry.unwrap().path()))
        }
        let read_dir = fs::read_dir("/hamlet/act1/../../..").unwrap();
        for entry in read_dir {
            out.push(format!("{:?}", entry.unwrap().path()))
        }
        let read_dir = fs::read_dir("/hamlet/act1/../../hamlet/act2/../act1/../../../").unwrap();
        for entry in read_dir {
            out.push(format!("{:?}", entry.unwrap().path()))
        }
        let f = fs::OpenOptions::new().write(true).open("/abc");

        if f.is_ok() {
            out.push("ROOT IS NOT SAFE".to_string());
        } else {
            out.push("ROOT IS SAFE".to_string());
        }

        out
    };

    for result in results {
        println!("{}", result);
    }
}
