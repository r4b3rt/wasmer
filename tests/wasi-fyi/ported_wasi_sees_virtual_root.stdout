"/.app"
"/.private"
"/bin"
"/dev"
"/etc"
"/fyi"
"/hamlet"
"/tmp"
"/hamlet/act1/../README.md"
"/hamlet/act1/../act1"
"/hamlet/act1/../act2"
"/hamlet/act1/../act3"
"/hamlet/act1/../act4"
"/hamlet/act1/../act5"
"/hamlet/act1/../bookmarks"
"/."
"//"
"/."
"//"
ROOT IS SAFE
